ui/src/shared/components/ScatterContainer.tsx

```diff
@@ -75,15 +86,17 @@ const ScatterContainer: FunctionComponent<Props> = ({
         onSetFillColumns(filteredGroupKeys)
       }
     }
+    if (!xColumn) {
```

Comment:
Is this to handle existing `XYViewProperties` that don't have an `xColumn` setting yet? Wowza I really wish we had the ability to migrate data...

I think if that is the case, there's no need to be setting the `xColumn` on the view properties by dispatching an effect, since those view properties will never be persisted. I think we can keep the x column resolution a purely render-path concern by declaring the `xColumn` could outside the effect like so:

```
const xColumn = storedXColumn || chooseXColumn(table)
```

It might help to extract that logic to a helper function and document it with a comment, so that we know we are dealing with an issue evolving data (like https://github.com/influxdata/influxdb/blob/e4772f06b238065e1bc25bc642042da73e926c4b/ui/src/shared/utils/vis.ts#L22-L32).